app/src/main/java/com/github/se/eduverse/ui/converter/PdfConverter.kt

```diff
@@ -223,6 +256,22 @@ fun OptionCard(testTag: String, optionName: String, icon: ImageVector, onClick:
       }
 }
 
+@Composable
+fun LoadingIndicator(onAbort: () -> Unit) {
+  Column(
+      modifier = Modifier.fillMaxSize(),
+      verticalArrangement = Arrangement.Center,
+      horizontalAlignment = Alignment.CenterHorizontally) {
+        CircularProgressIndicator(
+            color = Color.Gray,
+            trackColor = Color.Cyan,
+            strokeWidth = 10.dp,
+            strokeCap = StrokeCap.Round)
+        Spacer(modifier = Modifier.height(16.dp))
+        Button(onClick = onAbort) { Text("Abort PDF Generation") }
+      }
+}
+
 /**
  * Helper function to retrieve the file name from a the uri of a file
  *
```

Comment:
I'm guessing the changes in this file and other changes related to the converter come from an other PR so I won't review them here, but be careful not merging the PR before the other one

Reply:
Yes I'll do that